src/Stack/SDist.hs

```diff
@@ -246,6 +260,61 @@ dirsFromFiles dirs = Set.toAscList (Set.delete "." results)
       | Set.member x s = s
       | otherwise = go (Set.insert x s) (FP.takeDirectory x)
 
+-- | Check package in given tarball. This will log all warnings
+-- and will throw an exception in case of critical errors.
+--
+-- Note that we temporarily decompress the archive to analyze it.
+checkSDistTarball :: (MonadIO m, MonadMask m, MonadThrow m, MonadCatch m, MonadLogger m, MonadReader env m, HasEnvConfig env)
+  => Path Abs File -- ^ Absolute path to tarball
+  -> m ()
+checkSDistTarball tarball = withTempTarGzContents tarball $ \pkgDir' -> do
+    pkgDir  <- (pkgDir' </>) `liftM`
+        (parseRelDir . FP.takeBaseName . FP.takeBaseName . toFilePath $ tarball)
+    --               ^ drop ".tar"     ^ drop ".gz"
+    cabalfp <- getCabalFileName pkgDir
+    name    <- parsePackageNameFromFilePath cabalfp
+    config  <- getPackageConfig name
+    (gdesc, pkgDesc) <- readPackageDescriptionDir config pkgDir
+    $logInfo $ "Checking package '" <>
+               (T.pack . P.unPackageName . P.pkgName . PD.package $ pkgDesc) <>
```

Comment:
The `unPackageName` accessor has existed only since Cabal-1.22.

I ran into this when I tried to build stack using the stack-7.8.yaml.

Maybe there should be an additional Travis build using the stack-7.8.yaml?